hal/armv7a/mmu.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * ARMv7 Cortex-A Memory Management Unit (MMU)
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "mmu.h"
+
+
+static u32 ttl1[0x1000] __attribute__((section(".uncached_ddr"), aligned(0x4000)));
+
+
+static inline void mmu_invalTLB(void)
+{
+	__asm__ volatile (" \
+		dsb; \
+		mcr p15, 0, r1, c8, c7, 0; \
+		dsb; \
+		isb");
+}
+
+
+static inline void mmu_setTTBR0(addr_t addr)
+{
+	__asm__ volatile (" \
+		dsb; \
+		mov r1, #1; \
+		mcr p15, 0, r1, c2, c0, 2; \
+		mcr p15, 0, %0, c2, c0, 0; \
+		mcr p15, 0, %0, c2, c0, 1; \
+		dsb; \
+		isb"
+		:
+		: "r" (addr)
+		: );
+}
+
+
+static inline void mmu_setDACR(u32 val)
+{
+	__asm__ volatile (" \
+		dsb; \
+		mcr p15, 0, %0, c3, c0, 0; \
+		dsb; \
+		isb"
+		:
+		: "r" (val)
+		:);
+}
+
+
+void mmu_enable(void)
+{
+	__asm__ volatile (" \
+		dsb; \
+		mrc p15, 0, r1, c1, c0, 0; \
+		orr r1, r1, #1; \
+		mcr p15, 0, r1, c1, c0, 0; \
+		dsb; \
+		isb"
+	);
```

Comment:
```suggestion
		dsb; \
		isb"
		: : : "r1" );
```
Missing `r1` in clobbers.